app/models/person_escort_record.rb

```diff
@@ -40,4 +45,28 @@ def build_responses!
 
     self
   end
+
+  def section_progress
+    sections_to_responded.map do |section, responses|
+      {
+        key: section,
+        status: set_progress(responses),
+      }
+    end
+  end
+
+private
+
+  def sections_to_responded
+    @sections_to_responded ||= framework_responses.joins(:framework_question).select('DISTINCT responded, framework_questions.section').group_by(&:section)
+  end
+
+  def set_progress(responses)
+    responded = responses.pluck(:responded).uniq
+    if responded.include?(true)
+      responded.include?(false) ? PERSON_ESCORT_RECORD_IN_PROGRESS : PERSON_ESCORT_RECORD_COMPLETED
+    else
+      PERSON_ESCORT_RECORD_NOT_STARTED
+    end
```

Comment:
Maybe we could simplify the logic here, the methods all and any could help:
```
responses.all?
responses.any?(true/false)
```

Reply:
updated 👍